test/app_automate_driver_test.rb

```diff
@@ -0,0 +1,148 @@
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+
+    mocked_element = mock('element')
+    mocked_element.expects(:click)
+
+    driver.expects(:find_element).with(:id, "test_button").returns(mocked_element)
+
+    driver.click_element("test_button")
+  end
+
+  def test_click_element_locator
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION, :accessibility_id)
+
+    mocked_element = mock('element')
+    mocked_element.expects(:click)
+
+    driver.expects(:find_element).with(:accessibility_id, "test_button").returns(mocked_element)
+
+    driver.click_element("test_button")
+  end
+
+  def test_wait_for_element_defaults
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+
+    mocked_element = mock('element')
+    mocked_element.expects(:displayed?).returns(true)
+
+    driver.expects(:find_element).with(:id, "test_button").returns(mocked_element)
+
+    driver.wait_for_element("test_button")
+  end
+
+  def test_wait_for_element_locator
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION, :accessibility_id)
+
+    mocked_element = mock('element')
+    mocked_element.expects(:displayed?).returns(true)
+
+    driver.expects(:find_element).with(:accessibility_id, "test_button").returns(mocked_element)
+
+    driver.wait_for_element("test_button")
+  end
+
+  def test_send_keys_to_element_defaults
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+
+    mocked_element = mock('element')
+    mocked_element.expects(:send_keys).with("Test_text")
+
+    driver.expects(:find_element).with(:id, "test_text_entry").returns(mocked_element)
+
+    driver.send_keys_to_element("test_text_entry", "Test_text")
+  end
+
+  def test_send_keys_to_element_locator
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION, :accessibility_id)
+
+    mocked_element = mock('element')
+    mocked_element.expects(:send_keys).with("Test_text")
+
+    driver.expects(:find_element).with(:accessibility_id, "test_text_entry").returns(mocked_element)
+
+    driver.send_keys_to_element("test_text_entry", "Test_text")
+  end
+
+  def test_start_driver
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+
+    Appium::Driver.any_instance.expects(:start_driver)
+    Open3.expects(:popen2).with(LOCAL_TUNNEL_COMMAND)
```

Comment:
https://github.com/bugsnag/maze-runner/pull/54/files#diff-4f9dfd7642654cddb10bc7b8249a9294R95 (`status = wait.value`) isn't covered. To exercise it, I think you can change this line to something like:
```ruby
    waiter = mock('Process::Waiter', value: mock('Process::Status'))
    Open3.expects(:popen2).with(LOCAL_TUNNEL_COMMAND).yields(mock('stdin'), mock('stdout'), waiter)
```